### PR TITLE
fix get_python_version_from_image method

### PR DESCRIPTION
**Pull request by adi88d**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/45687)

## Description
- Update the methods to use `packaging.version.parse` for getting the python version and return string instead of float:
  - `get_python_version_from_image`
  - `get_python_exec`
  - `build_pylint_command`
  - `add_typing_module`
  - `_run_xsoar_linter`
  - `_docker_image_create`

- Update all the places using python version from float to string inside `demisto_sdk/commands/lint/`
- Update tests.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/lint/commands_builder.py [hunk 1]
```diff
@@ -20,17 +20,18 @@
...
     Args:
-        py_num(float): Python version X.Y
+        py_num(str): Python version X.Y
         is_py2(bool): for python 2 version, Set True if the returned result should have python2 or False for python.
 
     Returns:
         str: python executable
     """
-    if py_num < 3:
+    py_ver = float(py_num)
```
Comment (dorschw) [inline demisto_sdk/commands/lint/commands_builder.py:33]: how do we handle 3.10?

File: demisto_sdk/commands/lint/helpers.py [hunk 2]
```diff
@@ -189,7 +190,7 @@ def add_typing_module(lint_files: List[Path], python_version: float):
     back_lint_files: List[Path] = []
     try:
         # Add typing import if needed to python version 2 packages
-        if python_version < 3:
+        if float(python_version) < 3:
```
Comment (dorschw) [inline demisto_sdk/commands/lint/helpers.py:193]: What happens when the version is `3.10.1`? or do we not provide the revision number?

File: demisto_sdk/commands/lint/linter.py [hunk 3]
```diff
@@ -247,7 +247,7 @@ def _gather_facts(self, modules: dict) -> bool:
                         self._pkg_lint_status["errors"].append('Unable to parse test-requirements.txt in package')
             elif not self._facts["python_version"]:
                 # get python version from yml
-                pynum = 3.7 if (script_obj.get('subtype', 'python3') == 'python3') else 2.7
+                pynum = '3.7' if (script_obj.get('subtype', 'python3') == 'python3') else '2.7'
```
Comment (dorschw) [inline demisto_sdk/commands/lint/linter.py:250]: perhaps it's time to update this as well, if we're editing here

File: demisto_sdk/commands/lint/tests/command_builder_test.py [hunk 4]
```diff
@@ -7,7 +7,7 @@
 values = [[Path("file1.py")], [Path("file1.py"), Path("file2.py")]]
 
 
-@pytest.mark.parametrize(argnames="py_num , expected_exec", argvalues=[(3.7, 'python3'), (2.7, 'python')])
+@pytest.mark.parametrize(argnames="py_num , expected_exec", argvalues=[('3.7', 'python3'), ('2.7', 'python')])
```
Comment (dorschw) [inline demisto_sdk/commands/lint/tests/command_builder_test.py:10]: please add tests involving `3.10` and `3.10.1`

**Review by dorschw — COMMENTED** [review]
good job

File: demisto_sdk/commands/lint/commands_builder.py [hunk 5]
```diff
@@ -231,8 +233,14 @@ def build_pylint_command(files: List[Path], docker_version: Optional[float] = No
...
     # TODO: remove when pylint will update its version to support py3.9
-    if docker_version and docker_version >= 3.9:
-        disable.append('unsubscriptable-object')
+
+    if docker_version:
+        py_ver = parse(docker_version)
+        major = py_ver.major
+        minor = py_ver.minor
+
+        if major == 3 and minor == 9:
```
Comment (yaakovi) [inline demisto_sdk/commands/lint/commands_builder.py:242]: what about 3.10?
keep the logic of "above" 3.9

**Review by yaakovi — APPROVED** [review]

File: demisto_sdk/commands/lint/commands_builder.py (hunk 1 above)
Comment (adi88d) [reply demisto_sdk/commands/lint/commands_builder.py]: I updated this method too

File: demisto_sdk/commands/lint/helpers.py (hunk 2 above)
Comment (adi88d) [reply demisto_sdk/commands/lint/helpers.py]: I updated this method too

File: demisto_sdk/commands/lint/linter.py (hunk 3 above)
Comment (adi88d) [reply demisto_sdk/commands/lint/linter.py:252]: what exactly do you want to update here?

File: demisto_sdk/commands/lint/tests/command_builder_test.py (hunk 4 above)
Comment (adi88d) [reply demisto_sdk/commands/lint/tests/command_builder_test.py]: done

File: demisto_sdk/commands/lint/commands_builder.py (hunk 5 above)
Comment (adi88d) [reply demisto_sdk/commands/lint/commands_builder.py]: done

**Comment by adi88d** [comment]
> Please test it on the `content/ds-test-3-10` [branch](https://github.com/demisto/content/tree/ds-test-3-10), make sure linter runs (specifically, mypy) locally and on CI (change the [branch's sdk version](https://github.com/demisto/content/blob/ds-test-3-10/dev-requirements-py3.txt#L4) to use the sdk from this sdk branch to test the CI)

tested and pass, link to the build:
https://app.circleci.com/pipelines/github/demisto/content/138717/workflows/f7d218a6-61c2-4a2b-88ca-44fd7629ace5

**Review by dorschw — APPROVED** [review]
nice

**Pull request merged**